spark/src/main/scala/io/ddf/spark/util/SparkUtils.scala

```diff
@@ -286,4 +286,70 @@ object SparkUtils {
     val schema = SparkUtils.schemaFromDataFrame(df)
     manager.newDDF(manager, df, Array(classOf[DataFrame]), null, null, schema)
   }
+
+  /**
+   * This util function produces appropriate error message for SQL-based APIs based on the SparkSQL error message and the SQL
+   *
+   * @param sqlError SparkSQL org.apache.spark.sql.AnalysisException's message
+   * @param sql the query
+   * @param expressions list of expressions including columns
+   * @return
+   */
+  def sqlErrorToDDFError(sqlError: String, sql:String, expressions: List[String]): String = {
+    // Invalid expression
+    val InvalidExpressionJSQLParser = ".*Encountered \" \".\" \". \"\" at line 1, column (\\d+).*".r
+
+    // Invalid expression SparkSQL
+    val InvalidExpression = ".*cannot recognize input near .+ in expression specification; line 1 pos (\\d+).*".r
+
+    // Invalid character SparkSQL
+    val InvalidCharacter1 = ".*character '(.)' not supported here\n.*".r
+    val InvalidCharacter2 = ".*character '(.)' not supported here;.*".r
+
+    // non-existent column SparkSQL
+    // "cannot resolve 'substring' given input columns ; line 1 pos 7"
+    val NonexistentColumns = ".*cannot resolve '(.+)' given input columns.*".r
+
+    def handleInvalidCharacter(sqlError: String, char: String): String = {
+      var errorMessage = s"Expressions or columns containing invalid character $char: "
+      val exprs = expressions.filter(exp => exp.contains(char))
+      if (exprs.nonEmpty) {
```

Comment:
This is redundant as `mkString` on empty list will just return empty string and thus the `errorMessage` will be the same. We can save 3 line here :smile: